v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/models/PreparedStatementValueObject.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates.models;
+
+public class PreparedStatementValueObject<T> {
+  private String dataType;
+  private T value;
+
+  public PreparedStatementValueObject(String dataType, T value) {
+    this.dataType = dataType;
+    this.value = value;
+  }
+
+  public String getDataType() {
+    return dataType;
+  }
+
+  public void setDataType(String dataType) {
+    this.dataType = dataType;
+  }
+
+  public T getValue() {
+    return value;
+  }
+
+  public void setValue(T value) {
+    this.value = value;
+  }
+
+  // Override toString() for better readability when printing objects
+  @Override
+  public String toString() {
+    return "PreparedStatementValueObject{" + "key='" + dataType + '\'' + ", value=" + value + '}';
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) return true;
+    if (obj == null || getClass() != obj.getClass()) return false;
+    PreparedStatementValueObject<?> that = (PreparedStatementValueObject<?>) obj;
+    return dataType.equals(that.dataType) && value.equals(that.value);
+  }
+
+  @Override
+  public int hashCode() {
+    return 31 * dataType.hashCode() + (value != null ? value.hashCode() : 0);
```

Comment:
I might be a little worried on having a custom hash function, especially if this gets called on objects that get serialized and deduped by dataflow.

in case you use https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/2072/files#r1891134130, a standard hash function would be implemented for the class out of box.

Reply:
Ok. I did a bit of deep dive on the the code for reverse replication and I feel that this will get constructed and used on the fly for a record, so dataflow will not try to dedupe these elements.

Even if it's not used by dataflow in it's serde path, and we are required to implement `hash` for say a hash function, it would be less error prone and collision resistant if we avoid a custom hash to the extent possible.

Reply:
Address the same @VardhanThigle